src/Util/Lock.php

```diff
@@ -44,6 +46,25 @@ private static function connectMemcache() {
 	}
 
 	/**
+	 * @brief Creates a semaphore key
+	 *
+	 * @param string $fn_name Name of the lock
+	 *
+	 * @return ressource the semaphore key
+	 */
+	private static function semaphore_key($fn_name) {
```

Comment:
Concerning CamelCase: I always forget, sorry. But concerning the function names: A "create..." would be wrong, I think. When a function returns a value, I think that the function name should represent this.

A "CreateSemaphoreKey" would only be correct if a boolean value was returned, because then something like this looks good:

````
if (self::CreateSemaphoreKey(...)) {
  ....
}
````

I think that a function that returns a value should be named similar to a variable with the same content, so for me this is much more better:

```
$key = self::SemaphoreKey(...);
```